BUILD.bazel

```diff
@@ -0,0 +1,309 @@
+
+cc_library(
+    name = "common",
+    srcs = [
+        "src/common/filesystem.cc",
+        "src/common/logger.cc",
+        "src/common/maliput_abort_and_throw.cc",
+        "src/common/range_validator.cc",
+    ],
+    hdrs = [
+        "include/maliput/common/assertion_error.h",
+        "include/maliput/common/filesystem.h",
+        "include/maliput/common/logger.h",
+        "include/maliput/common/maliput_abort.h",
+        "include/maliput/common/maliput_copyable.h",
+        "include/maliput/common/maliput_deprecated.h",
+        "include/maliput/common/maliput_hash.h",
+        "include/maliput/common/maliput_never_destroyed.h",
+        "include/maliput/common/maliput_throw.h",
+        "include/maliput/common/maliput_unused.h",
+        "include/maliput/common/passkey.h",
+        "include/maliput/common/profiler.h",
+        "include/maliput/common/range_validator.h",
+    ],
+    copts = ["-std=c++17"],
```

Comment:
Should we here try to match arguments passed to C++ compiler when using cmake build? 

See [DefaultCFlags.cmake](https://github.com/maliput/maliput/blob/3dca2709446a25169f28cfeb6a5c4157a73ca2d1/cmake/DefaultCFlags.cmake) file

Reply:
Ah, yes...I wanted to come back to this - it scares me a little.

I added this to the TODO in the file

```
# TODO(stonier): dig into the cflags maliput sets in cmake/DefaultCFlags.cmake
# and verify they all work in a bazel environment. Be a little cautious and
# discerning - bazel packages are source packages, some of those flags
# may have ramifications for people on other platforms (e.g. -std=c++17
# immediately breaks if someone depends on maliput from windows) and/or
# compilers.
```

Also created an issue to work on this: #569